### PR TITLE
Feature/editor base to es6

### PR DESCRIPTION
**Pull request by iNewLegend**

### CODE REVIEW

File: assets/dev/js/editor/editor-base.js [hunk 1]
```diff
@@ -1,43 +1,44 @@
-/* global ElementorConfig */
```
Comment (matipojo) [inline assets/dev/js/editor/editor-base.js]: Why?

File: assets/dev/js/editor/editor-base.js [hunk 2]
```diff
@@ -881,16 +878,13 @@ const App = Marionette.Application.extend( {
...
 		} );
-	},
+	}
 
-	onPreviewLoadingError: function() {
-		const self = this;
-
-		const debugUrl = self.config.document.urls.preview + '&preview-debug',
+	onPreviewLoadingError() {
+		const self = this,
```
Comment (matipojo) [inline assets/dev/js/editor/editor-base.js]: Remove `self` from here and whole file.

**Review by matipojo — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/editor-base.js [hunk 3]
```diff
@@ -502,7 +497,7 @@ const App = Marionette.Application.extend( {
 			},
 			onConfirm: null,
 			onCancel: function() {
-				parent.history.go( -1 );
+				window.history.go( -1 );
```
Comment (matipojo) [inline assets/dev/js/editor/editor-base.js]: It's from the iframe, so it should go to parent's back.

**Review by matipojo — CHANGES_REQUESTED** [review]

File: tests/qunit/js/editor-base.spec.js [hunk 4]
```diff
@@ -1,67 +1,53 @@
-/* global ElementorConfig */
-import editorBase from './../../../assets/dev/js/editor/editor-base';
+import EditorBase from './../../../assets/dev/js/editor/editor-base';
 
-const App = editorBase.extend();
+export default class EditorBaseTest extends EditorBase {
```
Comment (matipojo) [inline tests/qunit/js/editor-base.spec.js]: Rename also the file name.
```suggestion
export default class EditorTest extends EditorBase {
```

File: tests/qunit/setup-editor.js [hunk 5]
```diff
@@ -1,4 +1,5 @@
-elementorCommon.ajax.send = ( action, options ) => {};
+elementorCommon.ajax.send = ( action, options ) => {
```
Comment (matipojo) [inline tests/qunit/setup-editor.js]: Why?

File: tests/qunit/setup-editor.js [hunk 6]
```diff
@@ -97,10 +97,61 @@ const elementorConfigSettings = {
...
+		},
+		settings: {
+			ui_theme: 'dark',
+			edit_buttons: '',
+			lightbox_in_editor: 'yes',
+		},
+	},
 };
 
-/* global jQuery */
```
Comment (matipojo) [inline tests/qunit/setup-editor.js]: Why?

File: tests/qunit/setup-editor.js [hunk 7]
```diff
@@ -153,16 +204,8 @@ Marionette.Region.prototype.attachHtml = () => {
...
 };
-
-fixture.setBase( 'tests/qunit' );
-
-QUnit.testStart( ( { module, name } ) => {
-	fixture.load( 'index.html' );
-} );
-
-QUnit.testDone( ( { module, name } ) => {
-	fixture.cleanup();
```
Comment (matipojo) [inline tests/qunit/setup-editor.js]: Why?

File: tests/qunit/tests.js [hunk 8]
```diff
@@ -1,4 +1,4 @@
-import elementor from './js/editor-base.spec.js';
+import EditorBase from './js/editor-base.spec.js';
```
Comment (matipojo) [inline tests/qunit/tests.js]: Rename

File: tests/qunit/mock/schemes/items.json [hunk 9]
```diff
@@ -0,0 +1,110 @@
+{
```
Comment (matipojo) [inline tests/qunit/mock/schemes/items.json]: Move to another commit

**Review by matipojo — CHANGES_REQUESTED** [review]

File: tests/qunit/setup-editor.js (hunk 6 above)
Comment (iNewLegend) [reply tests/qunit/setup-editor.js]: Not needed.
image

should be handled like this:
image

File: tests/qunit/setup-editor.js (hunk 7 above)
Comment (iNewLegend) [reply tests/qunit/setup-editor.js]: I added this manually without using fixture.

**Review by matipojo — DISMISSED** [review]

File: assets/dev/js/editor/editor-base.js [hunk 10]
```diff
@@ -903,14 +899,14 @@ const App = Marionette.Application.extend( {
...
-			self.showFatalErrorDialog( dialogOptions );
-		} ).fail( function( response ) { //Iframe can't be loaded
-			self.showFatalErrorDialog( {
+		jQuery.get( debugUrl, () => {
+			this.showFatalErrorDialog( dialogOptions );
+		} ).fail( ( response ) => { //Iframe can't be loaded
+			this.showFatalErrorDialog( {
 				className: 'elementor-preview-loading-error',
 				headerMessage: debugData.header,
 				message: response.statusText + ' ' + response.status + ' ' + previewDebugLink,
```
Comment (kobizz) [inline assets/dev/js/editor/editor-base.js]: Use arrow function

File: assets/dev/js/editor/editor-base.js [hunk 11]
```diff
@@ -938,10 +934,10 @@ const App = Marionette.Application.extend( {
 		};
```
Comment (kobizz) [inline assets/dev/js/editor/editor-base.js]: Use arrow function

**Review by kobizz — CHANGES_REQUESTED** [review]

File: tests/qunit/js/editor-test.js [hunk 12]
```diff
@@ -1,22 +1,21 @@
-import EditorBase from './../../../assets/dev/js/editor/editor-base';
+import EditorBase from 'elementor-editor/editor-base';
+import EditorElementor from 'elementor-editor/editor-elementor';
 
-export default class EditorTest extends EditorBase {
```
Comment (matipojo) [inline tests/qunit/js/editor-test.js]: On unit test the `EditorBase` should be used, without the UI from the real editor.

File: assets/dev/js/editor/editor-elementor.js [hunk 13]
```diff
@@ -0,0 +1,93 @@
+import EditorBase from './editor-base';
```
Comment (matipojo) [inline assets/dev/js/editor/editor-elementor.js]: Why do you need a separated file?

File: tests/qunit/js/editor-test.js [hunk 14]
```diff
@@ -60,12 +59,20 @@ export default class EditorTest extends EditorBase {
 		this.initPanel();
 
 		this.previewLoadedOnce = true;
+	}
 
-		// Do nothing, else.
+	onStart( options ) {
+		const NProgress = {
```
Comment (matipojo) [inline tests/qunit/js/editor-test.js]: Don't test e2e.

**Review by matipojo — CHANGES_REQUESTED** [review]

File: tests/qunit/js/editor-test.js [hunk 15]
```diff
@@ -0,0 +1,83 @@
...
+			this.$preview.trigger( 'load' );
+		} );
+
+		super.initPreview();
+	}
+
+	initFrontend() {
+		elementorFrontend.init = () => console.log( 'initFrontend' );
+
+		elementorFrontend.elements.$body = jQuery( '#elementor-fake' );
```
Comment (matipojo) [inline tests/qunit/js/editor-test.js]: Rename fake -> test.

File: tests/qunit/js/editor-test.js [hunk 16]
```diff
@@ -0,0 +1,83 @@
...
+
+		super.initFrontend();
+
+		// Shortcut bind, in other words make shortcuts listen to iframe.
+		elementorFrontend.elements.$window = jQuery( '#elementor-preview-iframe' );
+	}
+
+	onPreviewLoaded() {
+		this.$preview = jQuery( '#elementor-preview-iframe' );
```
Comment (matipojo) [inline tests/qunit/js/editor-test.js]: All common rows between EditorTest & Editor should be moved to EditorBase.

**Review by matipojo — CHANGES_REQUESTED** [review]

**Review by matipojo — APPROVED** [review]

**Review by kobizz — APPROVED** [review]
